src/lsst_efd_client/efd_helper.py

```diff
@@ -671,3 +868,382 @@ def _parse_schema(topic, schema):
+        client=None,
+    ):
+        self._influx_client = EfdClientTools.get_client(
+            efd_name,
+            EfdClientSync.mode,
+            db_name,
+            creds_service,
+            timeout,
+            client,
+        )
+
+    def get_topics(self):
+        topics = self._influx_client.do_query("SHOW MEASUREMENTS")
+        return topics["name"].tolist()
+
+    def get_fields(self, topic_name):
+        fields = self._influx_client.do_query(
+            f'SHOW FIELD KEYS FROM "{self._influx_client.db_name}"'
+            f'."autogen"."{topic_name}"'
+        )
+        return fields["fieldKey"].tolist()
+
+    @property
+    def query_history(self):
+        """Return query history
+
+        Returns
+        -------
+        results : `list`
+            All queries made with this client instance
+        """
+        return self._influx_client.query_history
+
+    def build_time_range_query(
+        self,
+        topic_name,
+        fields,
+        start,
+        end,
+        is_window=False,
+        index=None,
+        convert_influx_index=False,
+        use_old_csc_indexing=False,
+    ):
+        return EfdClientTools.build_time_range_query(
+            topic_name,
+            fields,
+            start,
+            end,
+            self._influx_client.db_name,
+            is_window,
+            index,
+            convert_influx_index,
+            use_old_csc_indexing,
+        )
+
+    def select_time_series(
+        self,
+        topic_name,
+        fields,
+        start,
+        end,
+        is_window=False,
+        index=None,
+        convert_influx_index=False,
+        use_old_csc_indexing=False,
+    ):
+        query = self.build_time_range_query(
+            topic_name,
+            fields,
+            start,
+            end,
+            is_window,
+            index,
+            convert_influx_index,
+            use_old_csc_indexing,
+        )
+        ret = self._influx_client.do_query(query, convert_influx_index)
+        if ret.empty and not self._is_topic_valid(topic_name):
+            raise ValueError(f"Topic {topic_name} not in EFD schema")
+        return ret
+
+    def select_top_n(
+        self,
+        topic_name,
+        fields,
+        num,
+        time_cut=None,
+        index=None,
+        convert_influx_index=False,
+        use_old_csc_indexing=False,
+    ):
+        query = EfdClientTools.build_select_top_n_query(
+            topic_name,
+            fields,
+            num,
+            self._influx_client.db_name,
+            time_cut=None,
+            index=None,
+            convert_influx_index=False,
+            use_old_csc_indexing=False,
+        )
+
+        ret = self._influx_client.do_query(query, convert_influx_index)
+
+        if ret.empty and not self._is_topic_valid(topic_name):
+            raise ValueError(f"Topic {topic_name} not in EFD schema")
+        return ret
+
+    async def select_packed_time_series(
```

Comment:
I think I'm missing something here, but why is this `async def` on the sync client?

Reply:
Yes, sorry!